File: src/2025-02-27-deepclaude37-rust/cpp/exercises/practice/phone-number/phone_number.cpp
#include "phone_number.h"
#include <cctype>
#include <stdexcept>

namespace phone_number {

phone_number::phone_number(const std::string& s) {
    std::string digits;
    for (char c : s) {
        if (std::isdigit(c)) {
            digits.push_back(c);
        }
    }

    if (digits.length() == 11 && digits[0] == '1') {
        digits = digits.substr(1);
    }

    if (digits.length() != 10) {
        throw std::domain_error("Invalid phone number");
    }

    if (digits[0] < '2' || digits[0] > '9') {
        throw std::domain_error("Area code must start with 2-9");
    }

    if (digits[3] < '2' || digits[3] > '9') {
        throw std::domain_error("Exchange code must start with 2-9");
    }

    num = digits;
}

}  // namespace phone_number

File: src/2025-03-16-08-42-59--a-deepclaude-full-python-0316-1/python/exercises/practice/phone-number/phone_number.py
class PhoneNumber:
    def __init__(self, number):
        # Clean the number by removing allowed punctuation
        cleaned = []
        for char in number:
            if char.isdigit():
                cleaned.append(char)
            elif char in ['+', ' ', '(', ')', '-', '.']:
                continue
            elif char.isalpha():
                raise ValueError("letters not permitted")
            else:
                raise ValueError("punctuations not permitted")
        
        cleaned_number = ''.join(cleaned)
        
        # Check length requirements
        if len(cleaned_number) < 10:
            raise ValueError("must not be fewer than 10 digits")
        if len(cleaned_number) > 11:
            raise ValueError("must not be greater than 11 digits")
        
        # Check if 11 digits starts with 1
        if len(cleaned_number) == 11:
            if cleaned_number[0] != '1':
                raise ValueError("11 digits must start with 1")
            cleaned_number = cleaned_number[1:]
        
        # Now we have a 10-digit number, validate area code and exchange code
        if cleaned_number[0] == '0':
            raise ValueError("area code cannot start with zero")
        if cleaned_number[0] == '1':
            raise ValueError("area code cannot start with one")
        
        if cleaned_number[3] == '0':
            raise ValueError("exchange code cannot start with zero")
        if cleaned_number[3] == '1':
            raise ValueError("exchange code cannot start with one")
        
        # Store the cleaned number and area code
        self.number = cleaned_number
        self.area_code = cleaned_number[:3]
    
    def pretty(self):
        """Format the phone number in a pretty format: (XXX)-XXX-XXXX"""
        return f"({self.area_code})-{self.number[3:6]}-{self.number[6:]}"
